src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -533,35 +533,37 @@ where
         let mut ticker = tokio::time::interval(timestamp_interval);
         ticker.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
 
-        let reclock_to_latest = dyncfgs::STORAGE_RECLOCK_TO_LATEST.get(&config.config.config_set());
+        let mut prev_binding_ts = mz_repr::Timestamp::MIN;
+        let mut prev_source_upper = Antichain::from_elem(FromTime::minimum());
```

Comment:
Initializing to the minimum values (rather than making these `Option`s) is fine here because (a) we'll never select 1970-01-01 as a binding ts anymore and (b) for the source upper we don't require strict increases. Not using `Option` makes things less explicit here, but avoids some unwrapping below so I thought I was worth it. No strong feelings though!

Reply:
I think we should keep the `Option<_>` here and not invent a null sentinel. It does require `unwraps` but we also want them to fire if this code evolves in the wrong way.

Reply:
Okay, brought the option back.